### PR TITLE
Feature demo ball c

### PR DESCRIPTION
**Pull request by Gian1080**

### CODE REVIEW

File: code/src/ball_c.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include <ball_c.hpp>
+
+void ball_c::update_ball()
+{
+        x_axis += direction_x;
+        y_axis += direction_y;
+
```
Comment (ThePatman3) [inline code/src/ball_c.cpp]: Why these empty lines?

File: code/headers/ball_c.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#include <stdint.h>
```
Comment (ThePatman3) [inline code/headers/ball_c.hpp:2]: This file lacks *#pragma once*. Is this intended?

**Review by ThePatman3 — CHANGES_REQUESTED** [review]
The code looks good, but I have left two comments of which at least the *#pragma once* one has to be resolved before I approve this pull request.

File: code/src/ball_c.cpp (hunk 1 above)
Comment (Gian1080) [reply code/src/ball_c.cpp]: For demo purposes I'd rather follow my own guidelines

File: code/headers/ball_c.hpp (hunk 2 above)
Comment (Gian1080) [reply code/headers/ball_c.hpp:2]: forgot, thanks!

Comment (Gian1080) [reply code/headers/ball_c.hpp:2]: repaired!

File: code/src/ball_c.cpp (hunk 1 above)
Comment (Gian1080) [reply code/src/ball_c.cpp]: ahh, missread comment, removed the empty lines, thanks!

**Review by ThePatman3 — APPROVED** [review]
All looks good now, although the header has no Doxygen. I will approve this nonetheless because this code is meant for a demo.

**Pull request merged**